riotctrl/tests/shell_json_test.py

```diff
@@ -0,0 +1,17 @@
+"""riotctrl.shell.json test module"""
+
+import riotctrl.shell.json
+
+
+def test_json_shell_interaction_parser():
+    """Test JSON parsing"""
+    parser = riotctrl.shell.json.JSONShellInteractionParser()
+
+    res = parser.parse('[{"test": [1234, {"obj": {"val": 3.14}}]}]')
```

Comment:
Could we add some trailing commas in... It would be good if the parser could ignore them.  Then writing json output has one less state to worry about.

Reply:
Mmmhh can do that, but I would prefer that if that does not work to stick with the current solution. I don't want to write my own json parser when Python has a perfectly fine (and fast) one on its own ;-).

Reply:
Sadly, this does not work (see [this run](https://github.com/RIOT-OS/riotctrl/actions/runs/501732479) for https://github.com/RIOT-OS/riotctrl/pull/8/commits/2b99f61bc1b8621690dbb70cec5a3b088c38a969

Reply:
Hmm I will try to find some more permissive parsers...

Reply:
From what I have seen we have:

- [json5](https://github.com/dpranke/pyjson5) which seems too slow but allows for trailing commas and comments
- [rapidjson](https://github.com/python-rapidjson/python-rapidjson/) which seems pretty fast and allows for everything (so far my favorite)
- [hyperjson](https://github.com/mre/hyperjson) which seems about as fast as rapidjson but I don't think supports trailing commas (it states it tries to be a drop in replacement)
- [pyyaml](https://github.com/yaml/pyyaml) Some say it is a hack but it allows the most permissive parsing... again pretty slow but at least official

Like I said, rapidjson is my favorite as it fits what we need and the incompatibilities are listed and don't seem to be a problem for what we will need.
```py
import rapidjson
rapidjson.loads('{"foo":"bar","baz":[1,2,3,],}', parse_mode=rapidjson.PM_TRAILING_COMMAS)
```

Reply:
Ok, instead of using `rapidjson` outright (as it is implemented using the CPython backend, making the code not as portable anymore), I provided it as an optional dependency, usable with the `RapidJSONShellInteractionParser`.